presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -136,7 +139,7 @@ private void addNewElement(int hashPosition, Block block, int position)
     {
         elementType.appendTo(block, position, elementBlock);
         if (elementBlock.getSizeInBytes() > FOUR_MEGABYTES) {
-            throw exceededLocalLimit(new DataSize(4, MEGABYTE));
+            throw new PrestoException(EXCEEDED_FUNCTION_MEMORY_LIMIT, String.format("the argument of %s exceeds of %s", functionName, MAX_FUNCTION_MEMORY));
```

Comment:
What about `The argument to array_distinct (or map-to-map cast) is too large. More than xMB of memory is needed to hold intermediate hash set`?